src/OperatingSystem.php

```diff
@@ -35,4 +41,21 @@ public static function onMac()
     {
         return PHP_OS === 'Darwin';
     }
+
+    /**
+     * Mac platform architecture.
+     *
+     * @return string
+     */
+    public static function macArchitectureId()
+    {
+        switch (php_uname('m')) {
+            case 'arm64':
+                return 'mac-arm';
+            case 'x86_64':
+                return 'mac-intel';
+            default:
+                return 'mac';
+        }
+    }
```

Comment:
This different from PR sent to `master`. Instead of having `onMacIntel()` and `onMacArm()` we can just stick with `onMac()` and then have a new method to return architecture value.